src/functions/ducklake_compaction_functions.cpp

```diff
@@ -404,14 +404,22 @@ DuckLakeCompactor::GenerateCompactionCommand(vector<DuckLakeCompactionFileEntry>
 		}
 	}
 
+	bool should_write_row_id = false;
+	if (type == CompactionType::REWRITE_DELETES) {
+		// when there are delete files, we always need to write row-ids because deleted rows create gaps
+		should_write_row_id = true;
+	} else if (type == CompactionType::MERGE_ADJACENT_TABLES) {
+		// if files are adjacent, we don't need to write the row-id to the file
+		should_write_row_id = !files_are_adjacent;
+	}
+
```

Comment:
Could we change this to a switch case?
Then if in the future another compaction method gets added we can get a nice error for the developer.